care/utils/evaluators/interpretation_evaluator.py

```diff
@@ -107,31 +107,35 @@ def evaluate_conditions(self, conditions, context):
                 metric_evaluator_obj = self.metric_cache[metric]
             else:
                 metric_evaluator = EvaluatorMetricsRegistry.get_evaluator(metric)
-                metric_evaluator_obj = metric_evaluator(
-                    context.get(metric_evaluator.context)
-                )
+                if metric_evaluator.context == "tag":
+                    metric_evaluator_obj = metric_evaluator(context)
+                else:
+                    metric_evaluator_obj = metric_evaluator(
+                        context.get(metric_evaluator.context)
+                    )
                 self.metric_cache[metric] = metric_evaluator_obj
             if not metric_evaluator_obj.apply_rule(
                 condition.get("operation"), condition.get("value")
             ):
                 return False
         return True
 
-    def get_matching_condition(self, context: dict, value: Any):
+    def get_matching_conditions(self, context: dict, value: Any):
+        matching_conditions = []
         for rule in self.rules:
             conditions = rule.get("conditions", {})
             if self.evaluate_conditions(conditions, context):
-                return rule, value
-        return None, None
+                matching_conditions.append(rule)
```

Comment:
We will only support the first one that was matched, Otherwise we will run into conflicts. If we need multiple interpretations, there should be another interpreter class built just for those types and it should be an option in each observation to choose which class is providing the interpretation.
That said, i dont see a case where you would need multiple interpretations.